ci/unit_tests/workspace_delete/main_test.py

```diff
@@ -0,0 +1,254 @@
+        deleteParams = list(self.deleteParamsBase)
+        deleteParams.extend(['-c', createOutputConfigPath, '-oc', deleteOutputConfigPath])
+        self.t_noException([deleteParams])
+
+        parser = argparse.ArgumentParser()
+        parser.add_argument('-c', '--common_configFilePaths', help='configuaration file', action='append')
+        parser.add_argument('-oc', '--common_output_config', help='output configuration file')
+        parser.add_argument('-cu','--conversation_url', required=False, help='url of the conversation service API')
+        parser.add_argument('-cv','--conversation_version', required=False, help='version of the conversation service API')
+        parser.add_argument('-cn','--conversation_username', required=False, help='username of the conversation service instance')
+        parser.add_argument('-cp','--conversation_password', required=False, help='password of the conversation service instance')
+        parser.add_argument('-cid','--conversation_workspace_id', required=False, help='workspace_id of the application.')
+        parser.add_argument('-wn','--conversation_workspace_name', required=False, help='name of the workspace')
+        parser.add_argument('-wnm','--conversation_workspace_match_by_name', required=False, help='true if the workspace name should be matched by name (or pattern if defined)')
+        parser.add_argument('-wnp','--conversation_workspace_name_pattern', required=False, help='regex pattern specifying a name of workspaces to be deleted')
+        parser.add_argument('-v','--verbose', required=False, help='verbosity', action='store_true')
+        args = parser.parse_args(deleteParams)
+        deleteInputConfig = Cfg(args)
+
+        workspaces = getWorkspaces(self.workspacesUrl, self.version, self.username, self.password)
+
+        # there should be no workspace with id specified in config file
+        workspacesFound = 0
+        for workspace in workspaces:
+            if workspace['workspace_id'] == getRequiredParameter(deleteInputConfig, 'conversation_workspace_id'):
+                workspacesFound += 1
+
+        assert workspacesFound == 0
+
+        # there should be still one workspace left (even with the same name)
+        assert len(workspaces) == 1
+
+        # check if workspace_id is not present in the output config
+        parser = argparse.ArgumentParser()
+        parser.add_argument('-c', '--common_configFilePaths', help='configuaration file', action='append')
+        args = parser.parse_args(['-c', deleteOutputConfigPath])
+        deleteOutputConfig = Cfg(args)
+
+        assert hasattr(deleteOutputConfig, 'conversation_workspace_id') == False
+
+    @pytest.mark.parametrize('envVarNameUsername, envVarNamePassword', [('WA_USERNAME', 'WA_PASSWORD')])
+    def test_deleteByName(self, envVarNameUsername, envVarNamePassword):
+        """Tests if workspace can be deleted by its id."""
+
+        workspaceName = 'deleteByName_workspace'
```

Comment:
Maybe we should try to delete multiple workspaces with same name (or if it's not possible then we should have test that checks that you can not do that)

Reply:
This is a good idea, done in b908224027fbce77cbcfeb534f1a5f76d08e7436